### PR TITLE
Limit the maximum value of unbinds in the connection dialog to signal_argument_size

### PR DESCRIPTION
**Pull request by Rindbee**
Previously, the range of `unbinds` was the default range of `SpinBox` (`[0, 100]`), now the maximum value is limited to the `size` of the parameter list of the **Signal**.

captrue

And the value of `unbinds` is taken into account when generating the parameter list of the **Receiver Method**.

I found `Script::has_script_signal` and `Script::get_script_signal_list`, but there is no `Script::get_script_signal`.

### CODE REVIEW

**Comment by KoBeWi** [comment]
The unbinds spinbox could get disabled if max is 0, so users wouldn't wonder why it's non-functional.

File: editor/connections_dialog.cpp [hunk 1]
```diff
@@ -365,6 +365,26 @@ void ConnectDialog::init(ConnectionData p_cd, bool p_edit) {
...
+			if (mi.name == signal) {
+				r_signal = mi;
+				break;
+			}
+		}
+	} else {
+		ClassDB::get_signal(source->get_class(), signal, &r_signal);
+	}
+
+	signal_argument_size = r_signal.arguments.size();
```
Comment (KoBeWi) [inline editor/connections_dialog.cpp:385]:
```suggestion
	int signal_argument_size = r_signal.arguments.size();
```
You could as well move the declaration here.

File: editor/connections_dialog.cpp [hunk 2]
```diff
@@ -619,6 +639,7 @@ void ConnectionsDock::_make_or_edit_connection() {
 	if (add_script_function) {
 		// Pick up args here before "it" is deleted by update_tree.
 		script_function_args = it->get_metadata(0).operator Dictionary()["args"];
+		script_function_args.resize(script_function_args.size() - cd.unbinds);
```
Comment (KoBeWi) [inline editor/connections_dialog.cpp:642]: What is this for? 🤔

Comment (Rindbee) [reply editor/connections_dialog.cpp:642]: For argument list of auto-generated functions (`Receiver Method` in script). If the number of parameters sent is not equal to the number of parameters of the receiving method, an error will be reported.

Comment (Rindbee) [reply editor/connections_dialog.cpp:642]: The unbinds spinbox will eventually decide how many parameters to unbind.
Its value is actually the unbind size of the newly constructed `callable`.
This error occurs if it is greater than 0 (the size of the arguments in `focus_entered` is `0`).
```
E 0:00:03:0853   emit_signalp: Error calling from signal 'focus_entered' to callable: VBoxContainer(test.gd)::_on_button_2_focus_entered : Method expected 0 arguments, but called with 0..
  <C++ Source>   core/object/object.cpp:1132 @ emit_signalp()
```

Comment (Rindbee) [reply editor/connections_dialog.cpp:642]: > The unbinds spinbox could get disabled if max is 0, so users wouldn't wonder why it's non-functional.

The maximum value is `0`, which means there are `no` parameters to unbind, that is, the size of the defined signal parameter list is `0`, disable it is reasonable.

**Comment by Rindbee** [comment]
For parameters in the final **Receiver Method** parameter list, if divided by where they come from, currently, they can be divided into 3 categories, as I mentioned in [A](https://github.com/godotengine/godot/pull/60143#issuecomment-1095404236) (there, I added another one, signal source, to increase flexibility of the connection dialog).

File: editor/connections_dialog.cpp (hunk 2 above)
Comment (KoBeWi) [reply editor/connections_dialog.cpp:641]: Ah, so this basically fixes a bug. Nice.

File: editor/connections_dialog.cpp [hunk 3]
```diff
@@ -365,6 +365,25 @@ void ConnectDialog::init(ConnectionData p_cd, bool p_edit) {
...
+				r_signal = mi;
+				break;
+			}
+		}
+	} else {
+		ClassDB::get_signal(source->get_class(), signal, &r_signal);
+	}
+
+	int signal_argument_size = r_signal.arguments.size();
+	unbind_count->set_max(signal_argument_size);
```
Comment (KoBeWi) [inline editor/connections_dialog.cpp:385]:
```suggestion
	unbind_count->set_max(r_signal.arguments.size());
```
Sorry for not noticing it earlier, but this variable is actually not necessary at all.

**Review by KoBeWi — APPROVED** [review]
Looks ok.

As for the disabling the SpinBox I mentioned earlier, when the method has 0 arguments, the max value of unbind is 0. The SpinBox can still be interacted with, but clicking or typing anything has no effect. It might be confusing to some people, so maybe it would be better to disable it explicitly. But maybe I'm wrong. It's not that important.

File: editor/connections_dialog.cpp (hunk 2 above)
Comment (Rindbee) [reply editor/connections_dialog.cpp:641]: > Ah, so this basically fixes a bug. Nice.

Yes, it does.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**